### PR TITLE
Address API review feedback

### PR DESCRIPTION
**Pull request by sima-zhu**
https://apiview.dev/Assemblies/Review/8ec9fe08197443d9be23718f8aad5276

### CODE REVIEW

File: sdk/search/azure-search-documents/swagger/readme.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ These settings apply only when `--tag=package-2019-05-searchindex-preview` is sp
 ``` yaml $(tag) == 'package-2019-05-searchindex-preview'
 namespace: com.azure.search.documents
 input-file:
-- https://github.com/Azure/azure-rest-api-specs/blob/master/specification/search/data-plane/Azure.Search/preview/2019-05-06-preview/searchindex.json
+- https://github.com/Azure/azure-rest-api-specs/blob/master/specification/search/data-plane/Azure.Search/preview/2019-05-06-preview/searchindex.jsonS
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/swagger/readme.md:64]: Typo here that will need to be fixed.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexAsyncClient.java [hunk 2]
```diff
@@ -542,6 +542,15 @@ SearchClientBuilder getSearchClientBuilder(String indexName) {
         }
     }
 
+    /**
+     * Lists all synonym map names for an Azure Cognitive Search service.
+     *
+     * @return a reactive response emitting the list of synonym map names.
+     */
+    public PagedFlux<String> listSynonymMapNames() {
```
Comment (srnagar) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexAsyncClient.java:550]: All APIs in this client are missing @ServiceMethod annotation. Also, add codesnippets where appropriate.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchClient.java [hunk 3]
```diff
@@ -229,15 +229,6 @@ public IndexDocumentsResult deleteDocuments(Iterable<?> documents) {
         return asyncClient.deleteDocumentsWithResponse(documents, context).block();
     }
 
-    /**
```
Comment (srnagar) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchClient.java:232]: The sync client APIs are missing @ServiceMethod annotation and codesnippets too.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexAsyncClient.java (hunk 2 above)
Comment (sima-zhu) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexAsyncClient.java:550]: I will add @ServiceMethod. 

Can improve JavaDoc and other documents in preview 5.

**Pull request merged**